File: sample/javaSample/src/main/java/jp/co/geniee/samples/SharedPreferenceManager.java
package jp.co.geniee.samples;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Set;

public class SharedPreferenceManager {

    public final static String SINGLE_BANNER_ZONE_ID = "SINGLE_BANNER_ZONE_ID";
    public final static String MULTIPLE_BANNERS_ZONE_ID = "MULTIPLE_BANNERS_ZONE_ID";
    public final static String NATIVE_AD_ZONE_ID = "NATIVE_AD_ZONE_ID";
    public final static String INTERSTITIAL_AD_ZONE_ID = "INTERSTITIAL_AD_ZONE_ID";
    public final static String VAST_AD_ZONE_ID = "VAST_AD_ZONE_ID";
    public final static String REWARDED_VIDEO_AD_ZONE_ID = "REWARDED_VIDEO_AD_ZONE_ID";

    private static SharedPreferenceManager sprefMrgSingleton;
    private Context mContext;
    private SharedPreferences sharedPreference;
    private SharedPreferences.Editor editor;

    private SharedPreferenceManager(Context context) {
        mContext = context;
        sharedPreference = PreferenceManager.getDefaultSharedPreferences(mContext);
        editor = sharedPreference.edit();
    }

    public static SharedPreferenceManager getInstance(Context context) {
        if (sprefMrgSingleton == null) {
            sprefMrgSingleton = new SharedPreferenceManager(context);
        }
        return sprefMrgSingleton;
    }

    public void putStringSet(String key, Set<String> set) {
        editor.putStringSet(key, set);
        editor.apply();
    }

    public Set<String> getStringSet(String key) {
        return sharedPreference.getStringSet(key, null);
    }

    public void putString(String key, String value) {
        editor.putString(key, value);
        editor.apply();
    }

    public String getString(String key) {
        return sharedPreference.getString(key, "");
    }

    public void putInt(String key, int value) {
        editor.putInt(key, value);
        editor.apply();
    }

    public int getInt(String key) {
        return sharedPreference.getInt(key, 0);
    }

    public void putFloat(String key, float value) {
        editor.putFloat(key, value);
        editor.apply();
    }

    public float getFloat(String key) {
        return sharedPreference.getFloat(key, 0);
    }

    public void putBoolean(String key, boolean value) {
        editor.putBoolean(key, value);
        editor.apply();
    }

    public boolean getBoolean(String key) {
        return sharedPreference.getBoolean(key, false);
    }

}

File: sample/UseLocalLibrary/GNAdSamples/kotlinSample/src/main/java/jp/co/geniee/samples/SharedPreferenceManager.kt
package jp.co.geniee.samples

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class SharedPreferenceManager private constructor(context: Context) {
    private val sharedPreference: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
    private val editor: SharedPreferences.Editor = sharedPreference.edit()

    fun putStringSet(key: String, set: Set<String>) {
        editor.putStringSet(key, set)
        editor.apply()
    }

    fun getStringSet(key: String): Set<String>? {
        return sharedPreference.getStringSet(key, null)
    }

    fun putString(key: String, value: String) {
        editor.putString(key, value)
        editor.apply()
    }

    fun getString(key: String): String {
        return sharedPreference.getString(key, "")
    }

    fun putInt(key: String, value: Int) {
        editor.putInt(key, value)
        editor.apply()
    }

    fun getInt(key: String): Int {
        return sharedPreference.getInt(key, 0)
    }

    fun putFloat(key: String, value: Float) {
        editor.putFloat(key, value)
        editor.apply()
    }

    fun getFloat(key: String): Float {
        return sharedPreference.getFloat(key, 0f)
    }

    fun putBoolean(key: String, value: Boolean) {
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getBoolean(key: String): Boolean {
        return sharedPreference.getBoolean(key, false)
    }

    companion object {

        const val SINGLE_BANNER_ZONE_ID = "SINGLE_BANNER_ZONE_ID"
        const val MULTIPLE_BANNERS_ZONE_ID = "MULTIPLE_BANNERS_ZONE_ID"
        const val NATIVE_AD_ZONE_ID = "NATIVE_AD_ZONE_ID"
        const val INTERSTITIAL_AD_ZONE_ID = "INTERSTITIAL_AD_ZONE_ID"
        const val VAST_AD_ZONE_ID = "VAST_AD_ZONE_ID"
        const val REWARDED_VIDEO_AD_ZONE_ID = "REWARDED_VIDEO_AD_ZONE_ID"

        private var sprefMrgSingleton: SharedPreferenceManager? = null

        fun getInstance(context: Context): SharedPreferenceManager {
            if (sprefMrgSingleton == null) {
                sprefMrgSingleton = SharedPreferenceManager(context)
            }
            return sprefMrgSingleton!!
        }
    }

}
